### ISSUE TITLE
DatePipe treat Date as DateTime

### ISSUE THREAD

**Issue by vinagreti** [angular/angular#12373]
**I'm submitting a ...**  (check one with "x")

```
[X] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
In (BRST) locale, when using `DatePipe` to format a `Date` string the output is the previous day and not the day I used.

**Expected behavior**
We expect that when using DatePipe to format a `Date` string the output should be the same date with only the format changed.

**Minimal reproduction of the problem with instructions**
Just run this plnker and look at the two dates showed. The first one show the wrong formated date. The second one shows the same date with the right format. If you look at the variables you will see that the difference between the first and the second one is only an space at the end of the string.
I think that space makes the pipe think that the input is in `Date` format but the first Date, without the space, is processed as DateTime in 00:00:00 time and as I'm using (BRST -3h) the utput shows the previous day.
https://embed.plnkr.co/Bn5TDukSd5qXDAIYNYdr/

**What is the motivation / use case for changing the behavior?**

The application is showing wrong dates.

**Please tell us about your environment:**

Ubuntu 16.04.1
Atom
NPM@3.5.2
Angular-cli@2.0.0-alpha.14
- **Angular version:** 2.0.X
  angular@2.1.0
- **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]
  
  
  
  Chrome Version 53.0.2785.116 (64-bit)
  Firefox Version 49
- **Language:** [all | TypeScript X.X | ES6/7 | ES5]
  all
- **Node (for AoT issues):** `node --version` =  
  Node 4.2.6

**Comment by DzmitryShylovich** [comment]
http://prntscr.com/cvwz6y Don't see any difference.

**Comment by pkozlowski-opensource** [comment]
@DzmitryShylovich I would assume that this is time-zone related so this is why you might not see it on your end. But it needs further investigation indeed.

**Comment by DzmitryShylovich** [comment]
@vinagreti can u add screenshot?

**Comment by vinagreti** [comment]
Look at the bottom right to see how my new Date() outpus and see what is your output. I think for those who are in UTC locale, the problem does not occur.
http://prntscr.com/cvx3h9

**Comment by vinagreti** [comment]
@DzmitryShylovich, do you think that triming the string solves the problem?

If you look my code, using space is the workaround for me to get the right date and if you remove it using the trim, neither my workaround or your PR will work for my case, don't you think?

**Comment by DzmitryShylovich** [comment]
can u paste screenshot without space?

**Comment by vinagreti** [comment]
The screenshot has the two inputs with and without space. The `date1` have no space at the and and the `date2` has space a the end.

http://prntscr.com/cvx3h9

**Comment by DzmitryShylovich** [comment]
@pkozlowski-opensource Intl doesn't support time zones thus we can do nothing on angular side.

**Comment by vinagreti** [comment]
With some more tests I found that the problem is with the `TimezoneOffset` used by `Date()` to calculate the point in the time.
But, as I am piping  a `Date` string and not a `DateTime` string, I think Angular 2 could use some hack to remove the `TimezoneOffset` to give us the right `Date`.

Angular 1 used to act like this, you can see it in this plunker: https://plnkr.co/edit/zbaT1YB5ddRMiDfDtaTS?p=preview

Something like:

```js
var data = '2013-10-26';
data = new Date(data.replace(/-/g, ','));
```

or

```js
var data = '2013-10-26';
data = new Date(data.getTime()+(data.getTimezoneOffset()*60*60*60));
```

**Comment by DzmitryShylovich** [comment]
so it works as expected in ng1 but fails in ng2?

**Comment by vinagreti** [comment]
I've created a PR #12380 that fix this problem.

**Comment by pkozlowski-opensource** [comment]
#12380 landed via 2aba8b0ff2c5d6979802b96842aceccad3d5716d so hopefully this issue is fixed now. @vinagreti I'm going to close this one based on the above but feel free to re-open if you still hitting problems.
